zabbix/pgsql/schema.sls

```diff
@@ -8,34 +8,23 @@
 {% set dbuser = settings.get('dbuser', defaults.dbuser) -%}
 {% set dbpassword = settings.get('dbpassword', defaults.dbpassword) -%}
 
-{% set dbroot_user = settings.get('dbroot_user') -%}
-{% set dbroot_pass = settings.get('dbroot_pass') -%}
-
-{% set sql_file = settings.get('sql_file', '/usr/share/doc/zabbix-server-pgsql/create.sql.gz') -%}
-
-# Connection args required only if dbroot_user and dbroot_pass defined.
-{% set connection_args = {} -%}
-{% if dbroot_user and dbroot_pass -%}
-{%  set connection_args = {'runas': 'nobody', 'host': dbhost, 'user': dbroot_user, 'password': dbroot_pass} -%}
-{% endif -%}
-
-# Check is there any tables in database.
-# salt.postgres.psql_query return empty result if there is no tables or 'False' on any error i.e. failed auth.
-{% set list_tables = "SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname != 'pg_catalog' AND schemaname != 'information_schema' LIMIT 1;" %}
-{% set is_db_empty = True -%}
-{% if salt.postgres.psql_query(query=list_tables, maintenance_db=dbname, **connection_args) -%}
-{%  set is_db_empty = False -%}
-{% endif -%}
+{% set sql_file = settings.get('sql_file', defaults.sql_file) -%}
 
 include:
   - zabbix.pgsql.conf
 
+# Check is there any tables in database.
+# returns changed if there are zero tables in the db
 check_db_pgsql:
-  test.configurable_test_state:
-    - name: Is there any tables in '{{ dbname }}' database?
-    - changes: {{ is_db_empty }}
-    - result: True
-    - comment: If changes is 'True' data import required.
+  cmd.run:
+    - name: "[[ $(psql -X -A -t -c \"SELECT count(tablename) FROM pg_catalog.pg_tables WHERE schemaname != 'pg_catalog' AND schemaname != 'information_schema';\" || echo \"-1\") -eq \"0\" ]] && echo \"changed=yes comment='DB needs schema import.'\" || echo \"changed=no comment='No DB import needed or possible.'\""
```

Comment:
Sorry I currently don't have time to test this solution. Also I'm not sure that adding more shell code is a proper direction of formula development. I don't  want to say that current solution is perfect, but there must be better ways to solve your issues.

Reply:
i will split it off to another merge request for further discussion so at least the other fixes can be merged, though it definitely is a bug. you cannot currently use the formula on a clean host (or in some pipeline that checks it), because the salt.postgeres.psql_quesry sould NOT be used in jinja, because that is compiled BEFORE anything else, no matter what includes.

i've tried experimenting with unless/onlyif, slots etc, but I didnn't figure how to process the output. eg this code whould theoretically work, if i somehow managed to process the string (__slot__ output) to decide True/False on that

```
check_db_pgsql:
  test.configurable_test_state:
    - name: Is there any tables in '{{ dbname }}' database?
    - changes: __slot__:salt:postgres.psql_query("{{ list_tables }}", maintenance_db="{{ dbname }}", host="{{ dbhost }}", user="{{ dbuser }}", password="{{ dbpassword }}")
    - result: True
    - comment: If changes is 'True' data import required.
```

Reply:
split off to #151